src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java

```diff
@@ -1255,7 +1255,21 @@ private JSONObject getDataFromRestCallPost(GitHubParsed gitHubParsed, GitHubRepo
         if (StringUtils.isNotEmpty(settings.getGraphqlUrl())) {
             graphqlUrl = settings.getGraphqlUrl();
         }
-        ResponseEntity<String> response = makeRestCallPost(graphqlUrl, repo.getUserId(), password, personalAccessToken, query);
+        ResponseEntity<String> response;
+        int retryCount = 0;
+        while(true) {
+            try {
+                response = makeRestCallPost(graphqlUrl, repo.getUserId(), password, personalAccessToken, query);
+                break;
+            } catch (Exception e) {
```

Comment:
This section of the code needs to better thought and implemented.

Reply:
@rvema  added check to retry only if HTTPStatus code 502